packages/server/tests/AuthenticationSignUp.test.ts

```diff
@@ -39,6 +45,23 @@ test.group('Authentication sign up tests group', (group) => {
         const userNickname = internet.userName();
         const password = generateStrongPassword();
 
+        Mail.trap((message) => {
+            assert.deepEqual(message.to, [
+                {
+                    address: email,
+                },
+            ]);
+
+            assert.deepEqual(message.from, {
+                address: 'no-reply@adonisenprovence.com',
+            });
+
+            assert.equal(
+                message.subject,
+                `Welcome ${userNickname}, please verify your email !`,
+            );
+        });
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏I commented the line that sends the email in `signUp` method, and tests did not break. We should probably use `assert.plan()` to ensure that assertions are made?

But I do not understand how the test can work without a kind of `waitFor`, because Japa can not know when we send the email we want to assert on. We could send the email after we returned the HTTP response for instance.

Reply:
Interesting ! digging deeper

Reply:
I achieved to be using sinon.spy to verify that the spy is called on the test end